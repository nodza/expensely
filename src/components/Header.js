import React from 'react';
import { NavLink} from 'react-router-dom';

const Header = () => (
  <header>
    <h2>Expense.ly</h2>
    <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Add New</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
  </header>
);

export default Header;