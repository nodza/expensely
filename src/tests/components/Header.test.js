import React from 'react';
import { shallow } from 'enzyme';
// import { ReactShallowRenderer} from 'react-test-renderer/shallow';
import Header from '../../components/Header';

test('should render header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();


  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  // expect(wrapper.find('h2').text()).toBe('Expense.ly');
  // console.log(renderer.getRenderOutput());
});

