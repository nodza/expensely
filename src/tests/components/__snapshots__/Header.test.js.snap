// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h2>
    Expense.ly
  </h2>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Add New
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Help
  </NavLink>
</header>
`;
